src/main/java/org/elasticsearch/index/store/Store.java

```diff
@@ -383,6 +383,21 @@ public static MetadataSnapshot readMetadataSnapshot(Path indexLocation, ESLogger
     }
 
     /**
+     * Returns <code>true</code> iff the given location contains an index an the index
+     * can be successfully opened. This includes reading the segment infos and possible
+     * corruption markers.
+     */
+    public static boolean canOpenIndex(Path indexLocation) throws IOException {
+        try (Directory dir = new SimpleFSDirectory(indexLocation)) {
+            failIfCorrupted(dir, new ShardId("", 1));
+            Lucene.readSegmentInfos(dir);
+            return true;
+        } catch (Exception ex) {
```

Comment:
cat we add a trace log here with the exception? Thinking it might be helpful to be able to know why...